src/PrettyPrinting.jl

```diff
@@ -552,8 +552,8 @@ end
 #
 ################################################################################
 
-mutable struct printer
-   io::IO
+mutable struct printer{IOT <: IO}
```

Comment:
This makes all functions using this `printer` thing type-stable (as they eventually access the underlying IO)